File: java/BinarySearch/BinarySearch.java
package BinarySearch;

/**
 * @author Ali Azhari
 *
 */
public class BinarySearch {

	// the total running time is O(log n)

	public static boolean BS(int[] data, int target, int low, int high) {

		if (low > high)
			return false; // Not found
		else {

			int mid = (int) Math.floor((low + high) / 2);

			if (target == data[mid])
				return true;
			else {

				if (target > data[mid])
					return BS(data, target, mid + 1, high);
				else
					return BS(data, target, low, mid - 1);
			}
		}
	
	}

	public static void main(String[] args) {

		int data[] = { 1, 3, 5, 7, 9, 13 };

		for (int i = 0; i < 15; i++)
		System.out.println(i +" is " + (BS(data, i, 0, 5) == true ? "found" : "not found"));

	}

}

File: python/BinarySearch/BinarySearch.py
# ******************************************
#  Author : Ali Azhari 
#  Created On : Tue Apr 20 2019
#  File : BinarySearch.py
# *******************************************/

def BS(data, target, low, high):

    # The total running time is O(logn)
    
    if low > high:
        return False

    mid = (low + high) // 2
    if data[mid] == target:
        return True

    if target > data[mid]:
        return BS(data, target, mid + 1, high)
    else:
        return BS(data, target, low, mid -1)
    

def Main():
    data = [1, 3, 5, 7, 9, 13]
    for i in range(15):
        print(f"{i} is " + ( 'Found' if BS(data, i, 0, len(data)-1) == True else 'Not Found'))


if __name__ == "__main__":
    Main()